security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisRunParameters.java

```diff
@@ -0,0 +1,233 @@
+package com.powsybl.security;
+
+import com.powsybl.action.Action;
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.computation.local.LocalComputationManager;
+import com.powsybl.security.detectors.DefaultLimitViolationDetector;
+import com.powsybl.security.interceptors.SecurityAnalysisInterceptor;
+import com.powsybl.security.limitreduction.LimitReduction;
+import com.powsybl.security.monitor.StateMonitor;
+import com.powsybl.security.strategy.OperatorStrategy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * Parameters used in {@link SecurityAnalysisProvider#run} called in {@link SecurityAnalysis} API
+ * @author Laurent Issertial {@literal <laurent.issertial at rte-france.com>}
+ */
+public class SecurityAnalysisRunParameters {
+
+    private static final Supplier<LimitViolationDetector> DEFAULT_DETECTOR_SUPPLIER = DefaultLimitViolationDetector::new;
+    private static final Supplier<LimitViolationFilter> DEFAULT_FILTER_SUPPLIER = LimitViolationFilter::load;
+    private static final Supplier<SecurityAnalysisParameters> DEFAULT_SA_PARAMETERS_SUPPLIER = SecurityAnalysisParameters::load;
+    private static final Supplier<ComputationManager> DEFAULT_COMPUTATION_MANAGER_SUPPLIER = LocalComputationManager::getDefault;
+
+    private LimitViolationDetector detector;
+    private LimitViolationFilter filter;
+    /**
+     * Handle command execution
+     */
+    private ComputationManager computationManager;
+    /**
+     * Specific security analysis parameters
+     */
+    private SecurityAnalysisParameters securityAnalysisParameters;
+    /**
+     * List of interceptors notified at various step of the security analysis
+     */
+    private List<SecurityAnalysisInterceptor> interceptors = new ArrayList<>();
+    /**
+     * List of strategies applied to solve violations occurring after a contingency
+     */
+    private List<OperatorStrategy> operatorStrategies = new ArrayList<>();
+    /**
+     * List of actions taken according to {@link #operatorStrategies}
+     */
+    private List<Action> actions = new ArrayList<>();
+    /**
+     * StateMonitor that defines the branch bus and threeWindingsTransformer about which information will be written after security analysis
```

Comment:
It would be nice to know which information, where is it written? It's also confusing "after security analysis" seems to mean after all contingencies, but it's not. I think the sentence was yours @EtienneLt (at least you did implement StateMonitor... a long time ago!); could you try to rewrite this sentence?
```suggestion
     * List of StateMonitors that define the branches, buses and threeWindingsTransformers about which information will be written after security analysis
```